x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -100,12 +105,34 @@ export class SyncPrivateLocationMonitorsTask {
         MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE,
       ]);
 
+      const allPrivateLocations = await getPrivateLocations(soClient, ALL_SPACES_ID);
```

Comment:
I know this is not changed per se, but how many private locations are we talking about in the worst case? Since we are fetching all of them in memory